array-manipulation-2/src/chunk.js

```diff
@@ -0,0 +1,23 @@
+/* exported chunk */
+function chunk(array, size) {
+  const arrayOfArrays = [];
+  let i = 0;
+
+  while (i < array.length) {
+    const subArray = [];
+
+    if (array.length - i >= size) {
+      for (let j = 0; j < size; j++) {
+        subArray.push(array[i]);
+        i++;
+      }
+    } else {
+      for (i; i < array.length; i++) {
+        subArray.push(array[i]);
+      }
+    }
+
+    arrayOfArrays.push(subArray);
+  }
+  return arrayOfArrays;
+}
```

Comment:
We can actually accomplish this using only one loop! Check out the snippet below
```
function chunk(array, size) {
  const chunks = [];
  for (let i = 0; i < array.length; i++) {
    if (i % size === 0) {
      chunks.push([array[i]]);
    } else {
      const lastChunk = chunks[chunks.length - 1];
      lastChunk.push(array[i]);
    }
  }
  return chunks;
}
```